java-util/src/main/java/io/druid/java/util/common/concurrent/Execs.java

```diff
@@ -147,4 +151,30 @@ public void rejectedExecution(Runnable r, ThreadPoolExecutor executor)
         }
     );
   }
+
+  private static final LongAdder fjpWorkerThreadCount = new LongAdder();
```

Comment:
There was already a suggestion to use longaddr with some benchmarking. Unless there's support with data that this is a problem I'm not inclined to change it.

Reply:
That benchmark creates 100 threads, that is completely irrelevant for this use. 

Obviously performance is not a big deal here, again, because new threads are created rarely and the overhead of the thread creation itself dominates the overhead of calling any methods of either `AtomicLong` or `LongAdder`.

The problem is that the misuse of `LongAdder` makes the code more confusing. A reader who is aware that `LongAdder` is added to JDK to replace `AtomicLong` when high contention is expected, may wonder if this is the case here? Was it measured and Druid devs used `LongAdder` intentionally? If so, a lot of threads are going to be created here, why? Etc.

Reply:
I think a better argument here is that the code has a race condition as is.

```

    fjpWorkerThreadCount.increment();
    t.setName(StringUtils.nonStrictFormat(name, fjpWorkerThreadCount.longValue()));
```

should take the state after the increment instead of relying on longValue.

Since there is state management instead of pure accumulation it makes sense to have an atomic state class, like `AtomicLong`